plugin/utils/shell.py

```diff
@@ -86,6 +87,183 @@ def create_lods():
 	return msgs
 
 
+def generate_rig_edit():
+    """Automatic rig edit generator by NDP. Detects posed bones and automatically generates nodes and offsets them."""
+    msgs = []
+    #logging.info(f"Generating rig edit")
+    print("-------------------------------------------------------------")
+    print("Initiating rig edit")
+    
+    #Check if selected object is a valid armature
+    if bpy.context.active_object.type != 'ARMATURE':
+        #Object is not an armature. Cancelling.
+        print("No armature selected.")
+        msgs.append(f"No armature selected.")
+        return msgs
+    
+    #Store current mode
+    Mode = bpy.context.mode
+    
+    #Store number of edits
+    Edits = 0
+
+    #Force Switch to pose mode.
+    bpy.ops.object.mode_set(mode='POSE')
+    
+    #Get the armature
+    armature = bpy.context.object
+    
+    #Iterate over every bone in the armature.
+    for i in armature.pose.bones:
+        #Check if the bones have been scaled
+        #Update locations
+        bpy.context.view_layer.update()
+        
+        #Check that no bones are scaled
+        if i.scale != Vector((1,1,1)):
+            #Reset scale
+            #i.scale = Vector((1,1,1))
+            print("WARNING: "+str(i.name)+" HAD SCALE, CANCELLING")
+            msgs.append(f"Warning: {str(i.name)} had scale transforms. Reset scale for all bones and try again.")
+            return msgs
+        
+        #Find all bones with transforms.
+        if (i.location != Vector((0, 0, 0)) or i.rotation_quaternion != Quaternion((1,0,0,0)) or i.scale != Vector((1,1,1))) and i.name[:5] != "NODE_":
```

Comment:
more pythonic i.name.startswith("NODE_")

Reply:
changed to "and not p_bone.startswith("NODE_")